lib/datadog/tracing/trace_digest.rb

```diff
@@ -141,6 +141,37 @@ def initialize(
 
         freeze
       end
+
+      # Creates a copy of this object, modifying the provided fields.
+      # @param field_value_pairs [Hash<String>] the fields to be overwritten
+      # @return [TraceDigest] returns a copy of this object with the `field_value_pairs` modified
+      def merge(field_value_pairs)
+        # DEV: Because we want to sometimes freeze the values provided to `TraceDigest`, it's best
```

Comment:
non-blocking: Is there any reason to be concerned about performance here?

Reply:
It's not really performance, but more about maintainability.
This code here would have to be duplicated here: https://github.com/DataDog/dd-trace-rb/blob/8b6432a4cd51b9e59f6fef596d0ba3d0a44197c8/lib/datadog/tracing/trace_digest.rb#L122-L140
But it has to decide which fields can be used as is, or which ones need to be `dup.freeze`.

This is just to make sure we don't duplicate the logic and forget to copy some values but not others.